From: Carl Keinath <carl.keinath@gmail.com>
Date: Sat, 22 Nov 2025 00:18:30 +0100
Subject: Fix shader inclusion order

Sort the shader files when generating Shaders.hpp. This makes `reprotest
--variations=+fileordering` happy.

Origin: Debian
Forwarded: no
Author: Chow Loong Jin <hyperair@debian.org>
---
 scripts/generateShaderIncludes.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/generateShaderIncludes.sh b/scripts/generateShaderIncludes.sh
index c941903..306d7c2 100755
--- a/scripts/generateShaderIncludes.sh
+++ b/scripts/generateShaderIncludes.sh
@@ -12,7 +12,8 @@ echo '#pragma once' > ./src/render/shaders/Shaders.hpp
 echo '#include <map>' >> ./src/render/shaders/Shaders.hpp
 echo 'static const std::map<std::string, std::string> SHADERS = {' >> ./src/render/shaders/Shaders.hpp
 
-for filename in `ls ${SHADERS_SRC}`; do
+find "$SHADERS_SRC" -maxdepth 1 -type f | sort -fdi | while read -r filepath; do
+	filename=${filepath#"${SHADERS_SRC}"/}
 	echo "--	${filename}"
 	
 	{ echo -n 'R"#('; cat ${SHADERS_SRC}/${filename}; echo ')#"'; } > ./src/render/shaders/${filename}.inc
